### PR TITLE
[fix](inverted index) Fix match_regexp to correctly handle empty string patterns

### PR DESCRIPTION
**Pull request by zzzxl1993**
## Proposed changes

1. Handle empty strings consistently for match_phrase with and without an index.

### CODE REVIEW

**Comment by zzzxl1993** [comment]
run buildall

File: be/src/vec/functions/match.cpp [hunk 1]
```diff
@@ -406,15 +406,6 @@ Status FunctionMatchRegexp::execute_match(FunctionContext* context, const std::s
     VLOG_DEBUG << "begin to run FunctionMatchRegexp::execute_match, parser_type: "
                << inverted_index_parser_type_to_string(inverted_index_ctx->parser_type);
 
-    if (match_query_str.empty()) {
```
Comment (xiaokang) [inline be/src/vec/functions/match.cpp:409]: should consider other query

**Comment by zzzxl1993** [comment]
run buildall

**Review by csun5285 — APPROVED** [review]
LGTM

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]

**Pull request merged**